src/Connector.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {ICurvePool} from "./interfaces/ICurvePool.sol";
+import {IStrategyInterface} from "./interfaces/IStrategyInterface.sol";
+
+contract Connector {
+    using SafeERC20 for IERC20;
+
+    address public immutable xvault;
+
+    IStrategyInterface public immutable strategy;
+
+    ICurvePool public immutable curvePool;
+
+    IERC20 public immutable assetA;
+    IERC20 public immutable assetB;
+
+    constructor(address _xvault, address _strategy, address _assetA, address _assetB) { // @todo - add sanity checks
```

Comment:
Given the rest of the system is upgradeable we can make these strats upradeable as well since that's trade-off is already taken for the rest of the protocol